### PR TITLE
feature/AKD-137: Added Azure 'Scale Sets Autoscale Notifications Enabled' plugin and test cases

### PR DESCRIPTION
**Pull request by ali-imran7**

### CODE REVIEW

File: plugins/azure/virtualmachines/autoscaleNotificationsEnabled.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Scale Sets Autoscale Notifications Enabled',
+    category: 'Virtual Machines',
+    description: 'Ensures that Virtual Machine scale sets have autoscale notifications enabled',
+    more_info: 'Autoscale automatically creates new instances when certain metrics are surpassed, or can destroy instances that are being underutilized. Autoscale notifications should be enabled to know about the status of autoscale operation.',
+    recommended_action: 'Ensure that autoscale notifications are enabled for all Virtual Machine Scale Sets.',
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/autoscaleNotificationsEnabled.js:9]:
```suggestion
    recommended_action: 'Ensure that autoscale notifications are enabled for all Virtual Machine Scale Sets',
```

File: plugins/azure/virtualmachines/autoscaleNotificationsEnabled.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Scale Sets Autoscale Notifications Enabled',
+    category: 'Virtual Machines',
+    description: 'Ensures that Virtual Machine scale sets have autoscale notifications enabled',
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/autoscaleNotificationsEnabled.js:7]:
```suggestion
    description: 'Ensures that Virtual Machine scale sets have autoscale notifications enabled.',
```

File: plugins/azure/virtualmachines/autoscaleNotificationsEnabled.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                }
+            });
+
+            virtualMachineScaleSets.data.forEach(virtualMachineScaleSet => {
+                let autoScaleNotifications = {};
+                if (virtualMachineScaleSet.id &&
+                    asMap[virtualMachineScaleSet.id.toLowerCase()] &&
+                    asMap[virtualMachineScaleSet.id.toLowerCase()].notifications &&
+                    asMap[virtualMachineScaleSet.id.toLowerCase()].notifications.length) {
+                    autoScaleNotifications = asMap[virtualMachineScaleSet.id.toLowerCase()].notifications[0];
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/autoscaleNotificationsEnabled.js:64]: Shouldn't we loop all notifications?

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/azure/virtualmachines/autoscaleNotificationsEnabled.js (hunk 3 above)
Comment (ali-imran7) [reply plugins/azure/virtualmachines/autoscaleNotificationsEnabled.js:64]: lets discuss this

**Pull request merged**